lib/internal/source_map/source_map_cache.js

```diff
@@ -91,6 +107,9 @@ function maybeCacheSourceMap(filename, content, cjsModuleInstance, isGeneratedSo
     /\/[*/]#\s+sourceMappingURL=(?<sourceMappingURL>[^\s]+)/,
     content,
   );
+  if (sourceURL == null) {
```

Comment:
nit: personally I like being explicit about the comparison with `===`, as supporting `null` + `undefined` implies both values could happen (can they both happen?).